src/ReactTelephoneInput.js

```diff
@@ -110,6 +111,9 @@ var ReactTelephoneInput = React.createClass({
             this.props.onChange(this.state.formattedNumber, this.state.selectedCountry);
         }
     },
+    shouldComponentUpdate(nextProps) {
+        return !isEqual(nextProps, this.props);
```

Comment:
hey @anilreddykatta, I need your project knowledge. does your render() function produce the same DOM output given the same props/state? i.e. is your render() function pure? in this case we could just use [PureRenderMixin](https://facebook.github.io/react/docs/pure-render-mixin.html) and it'd be nice.